src/abstractdataframe/sort.jl

```diff
@@ -14,6 +14,24 @@
 #                  which allows a user to specify column specific orderings
 #                  with "order(column, rev=true, ...)"
 
+function _check_sort_args(lt, by, rev, order)
```

Comment:
Why not put this in function signatures instead? People should be used to the kind of `MethodError` that is printed. If we start checking the type of all arguments manually the codebase is going to get quite large. :-)

Reply:
Totally agreed. For some reason I started copying the old design. Now all kwargs have proper type restrictions.